File: Source/Services/RPSLS.PHPPlayer.Api/src/app/Services/PickStrategyFactory.php
<?php

namespace App\Services;

class PickStrategyFactory 
{
    private const _ROCK = "rock";
    private const _PAPER = "paper";
    private const _SCISSORS = "scissors";
    private const _LIZARD = "lizard";
    private const _SPOCK = "spock";
    private const _RANDOM = "random";
    private const _ITERATIVE = "iterative";

    private $_defaultStrategy = "random";

    public function getStrategy() 
    {
        switch($this->_defaultStrategy) 
        {
            case self::_ROCK:
                return new PickContext(new FixedStrategy(0));
            case self::_PAPER:
                return new PickContext(new FixedStrategy(1));
            case self::_SCISSORS:
                return new PickContext(new FixedStrategy(2));
            case self::_LIZARD:
                return new PickContext(new FixedStrategy(3));
            case self::_SPOCK:
                return new PickContext(new FixedStrategy(4));
            case self::_RANDOM:
                return new PickContext(new RandomStrategy());
            case self::_ITERATIVE:
                return new PickContext(new IterativeStrategy());
            default: 
                return new PickContext(new RandomStrategy());
        }
    }

    public function setDefaultStrategy($newDefaultStrategy)
    {
        if($newDefaultStrategy !== NULL)
        {
            $this->_defaultStrategy = strtolower($newDefaultStrategy);
        }
    }
}

File: Source/Services/RPSLS.DotNetPlayer.Api/Services/PickStrategyFactory.cs
using RPSLS.DotNetPlayer.Api.Models;
using RPSLS.DotNetPlayer.API.Strategies;

namespace RPSLS.DotNetPlayer.API.Services
{
    public class PickStrategyFactory : IPickStrategyFactory
    {
        private const string _ROCK = "rock";
        private const string _PAPER = "paper";
        private const string _SCISSORS = "scissors";
        private const string _LIZARD = "lizard";
        private const string _SPOCK = "spock";
        private const string _RANDOM = "random";
        private const string _ITERATIVE = "iterative";

        private string _defaultStrategy = _RANDOM;

        public IStrategy GetStrategy()
        {
            switch(_defaultStrategy)
            {
                case _ROCK:
                    return new FixedStrategy(RPSLSEnum.Rock);
                case _PAPER:
                    return new FixedStrategy(RPSLSEnum.Paper);
                case _SCISSORS:
                    return new FixedStrategy(RPSLSEnum.Scissors);
                case _LIZARD:
                    return new FixedStrategy(RPSLSEnum.Lizard);
                case _SPOCK:
                    return new FixedStrategy(RPSLSEnum.Spock);
                case _ITERATIVE:
                    return new IterativeStrategy();
                case _RANDOM:
                default:
                    return new RandomStrategy();
            }
        }

        public void SetDefaultStrategy(string newDefaultStrategy)
        {
            if(!string.IsNullOrEmpty(newDefaultStrategy))
            {
                _defaultStrategy = newDefaultStrategy.ToLower();
            }
        }
    }
}
